src/Console/ConfigurationResolver.php

```diff
@@ -588,10 +588,15 @@ private function parseRules()
             return $this->getConfig()->getRules();
         }
 
-        $jsonRules = json_decode($this->options['rules'], true);
+        $rules = trim($this->options['rules']);
 
-        if (is_array($jsonRules)) {
-            return $jsonRules;
+        if (in_array($rules[0], array('[', '{', '"', "'"), true)) {
+            $rules = json_decode($rules, true);
+            if (!is_array($rules)) {
```

Comment:
if it was decoded we already know it's an assoc object as it has to starts with `{` (due to prev comment).
then, consume `json_last_error` instead